src/test/store/receive-profile.test.js

```diff
@@ -307,36 +335,42 @@ describe('actions/receive-profile', function() {
       delete window.fetch;
     });
 
-    it('can retrieve a profile from the addon', async function() {
-      const { dispatch, getState } = setup();
-      await dispatch(retrieveProfileFromAddon());
-
-      const state = getState();
-      expect(getView(state)).toEqual({ phase: 'DATA_LOADED' });
-      expect(ProfileViewSelectors.getCommittedRange(state)).toEqual({
-        start: 0,
-        end: 1007,
+    for (const profileAs of ['json', 'arraybuffer', 'gzip']) {
+      describe('receive profile as ' + profileAs, function() {
+        it('can retrieve a profile from the addon', async function() {
+          const { dispatch, getState } = setup(profileAs);
+          await dispatch(retrieveProfileFromAddon());
+
+          const state = getState();
+          expect(getView(state)).toEqual({ phase: 'DATA_LOADED' });
+          expect(ProfileViewSelectors.getCommittedRange(state)).toEqual({
+            start: 0,
+            end: 1007,
+          });
+          expect(ProfileViewSelectors.getProfile(state).threads).toHaveLength(
+            3
+          ); // not empty
+        });
+
+        it('tries to symbolicate the received profile', async () => {
+          const { dispatch, geckoProfiler } = setup();
```

Comment:
I think you wanted this here:
```suggestion
          const { dispatch, geckoProfiler } = setup(profileAs);
```
Or keep this test out of the loop. Both works for me, but my preference would be to keep it out of the loop as I don't see the advantage of running it through all ways ofd receiving the profile.

Reply:
Fixed! I moved the test out of the loop (and simplified stuff a bit).